src/sentry/api/endpoints/sudo.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import absolute_import
+
+import json
+
+from django.contrib import auth
+from rest_framework.response import Response
+from sudo.utils import grant_sudo_privileges
+
+from sentry.api.base import Endpoint
+from sentry.models import Authenticator
+
+
+class SudoEndpoint(Endpoint):
+    permission_classes = ()
+
+    def post(self, request):
+        authenticated = False
+
+        if 'challenge' in request.DATA and 'response' in request.DATA:
+            try:
+                interface = Authenticator.objects.get_interface(request.user, 'u2f')
+                if not interface.is_enrolled:
+                    raise LookupError()
+
+                challenge = json.loads(request.DATA['challenge'])
+                response = json.loads(request.DATA['response'])
+                authenticated = interface.validate_response(request, challenge, response)
+            except ValueError:
```

Comment:
You can catch both of these at the same time with `except (ValueError, LookupError)`

The only real comment I have here though is that this is potentially catching a wider group of errors than you'd want or expect. I'm not sure within this what you expect to trigger a `ValueError` either. And the concern here is that `LookupError` covers a lot of stuff. Including both `KeyError` and `AttributeError`.

Reply:
I've updated to catch `LookupError` if u2f interface does not exist or user is not enrolled, and to catch `LookupError` only when decoding json.